core/app/services/spree/account/addresses/base.rb

```diff
@@ -0,0 +1,39 @@
+module Spree
+  module Account
+    module Addresses
+      class Base
+        prepend Spree::ServiceModule::Base
+
+        private
+
+        def fill_country_and_state_ids(params)
+          replace_country_iso_with_id(params)
+          fill_state_id(params)
+        end
+
+        def replace_country_iso_with_id(params)
+          iso = params[:country_iso]
+          return params unless iso.present?
+
+          params[:country_id] = Spree::Country.by_iso(iso)&.id
```

Comment:
@kgorazd let's set `country` as an attribute of this class so we can refer to it later in `fill_state_id`